cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -167,7 +167,8 @@ renderShelleyQueryCmdError err =
       Text.pack $ "The stake pool associated with: " <> nodeOpCert <> " was not found. Ensure the correct KES key has been " <>
                   "specified and that the stake pool is registered. If you have submitted a stake pool registration certificate " <>
                   "in the current epoch, you must wait until the following epoch for the registration to take place."
-
+    ShelleyQueryCmdDecodeError typeName decoderError ->
```

Comment:
Rather than propagate `Text` let's rename the error constructor to `ShelleyQueryCmdPoolStateDecodeError` and write a specific error message for the pool state decoding failure.

Reply:
Done!